coderd/database/dbpurge/dbpurge_test.go

```diff
@@ -177,57 +195,75 @@ func TestDeleteOldWorkspaceAgentLogs(t *testing.T) {
 	t.Run("AgentHasNotConnectedSinceWeek_LogsExpired", func(t *testing.T) {
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
 		defer cancel()
+		clk := quartz.NewMock(t)
+		clk.Set(now).MustWait(ctx)
 
-		// given
-		agent := mustCreateAgentWithLogs(ctx, t, db, user, org, tmpl, tv, now.Add(-8*24*time.Hour), t.Name())
+		// After dbpurge completes, the ticker is reset. Trap this call.
+		trapReset := clk.Trap().TickerReset()
+		defer trapReset.Close()
```

Comment:
This seems shared between all tests, how about refactoring to a helper like below, or consider only testing it in one test and assuming it applies to the rest?

```go
	t.Helper()

	// After dbpurge completes, the ticker is reset. Trap this call.
	trapReset := clk.Trap().TickerReset()
	defer trapReset.Close()

	closer := do(clk)
	
	// Wait for the initial nanosecond tick.
	clk.Advance(time.Nanosecond).MustWait(ctx)

	trapReset.MustWait(ctx).Release() // Wait for ticker.Reset()

	return closer
```

Perhaps this example can even be taken further, just here as food for thought.

Reply:
I modify this test in the follow-up PR https://github.com/coder/coder/pull/14460 so this won't be an issue.